### PR TITLE
add tests, fix issues

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: src/mock_miner.cpp [hunk 1]
```diff
@@ -167,8 +162,20 @@ Publications MockMiner::mine(const PublicationData& publicationData,
...
   ATV atv = generateValidATV(publicationData, lastKnownVbkBlockHash, state);
-  VTB vtb = generateValidVTB(
-      atv.containingBlock, lastKnownBtcBlockHash, vbkBlockDelay, state);
+
+  for (uint32_t i = 0; i != vbkBlockDelay; ++i) {
+    BlockIndex<vbk_block_t>* tip = vbk_blockchain->getBestChain().tip();
+
+    assert(tip != nullptr);
+
+    VbkBlock mindeBlock = vbk_miner->createNextBlock(*tip, uint128());
```
Comment (blackyblack) [inline src/mock_miner.cpp]: minedBlock

File: src/storage/endorsements_repository_inmem.cpp [hunk 2]
```diff
@@ -37,13 +37,16 @@ std::vector<Endorsement> EndorsementsRepositoryInmem::getEndorsementsInChain(
...
+  try {
+    for (const auto& endorsement : endorsed_map.at(endorsedBlockHash)) {
+      for (const auto& containingHashes : endorsement.containingVbkHashes) {
+        if (func(containingHashes)) {
+          result.push_back(endorsement);
+          break;
+        }
       }
     }
+  } catch (const std::out_of_range&) {
```
Comment (blackyblack) [inline src/storage/endorsements_repository_inmem.cpp]: Why do we hide an exception here?

File: test/blockchain/pop/pop_vbk_block_tree_test.cpp [hunk 3]
```diff
@@ -41,9 +100,121 @@ struct VbkBlockTreeTestFixture : ::testing::Test {
...
+  uint32_t numVbkBlocks = 200;
+
+  mock_miner->mineVbkBlocks(numVbkBlocks, state);
+  ASSERT_TRUE(state.IsValid());
+
+  // initially we have to endorse the genesis block to update the vbkTest chain
+  // with the 200 context blocks from the mock_miner state
+  endorseVtbBlock(0);
+
+  ASSERT_TRUE(uint32_t(vbkTest->getBestChain().size() - 2) == numVbkBlocks);
```
Comment (blackyblack) [inline test/blockchain/pop/pop_vbk_block_tree_test.cpp]: Why vbkTest chain is 2 blocks longer than mined in previous step?

File: src/mock_miner.cpp (hunk 1 above)
Comment (Mr-Leshiy) [reply src/mock_miner.cpp]: fixed

File: test/blockchain/pop/pop_vbk_block_tree_test.cpp (hunk 3 above)
Comment (Mr-Leshiy) [reply test/blockchain/pop/pop_vbk_block_tree_test.cpp]: fixed

File: src/storage/endorsements_repository_inmem.cpp (hunk 2 above)
Comment (Mr-Leshiy) [reply src/storage/endorsements_repository_inmem.cpp]: fixed

**Pull request merged**